src/transformers/training_args.py

```diff
@@ -802,6 +812,9 @@ def __post_init__(self):
                 "Mixed precision training with AMP or APEX (`--fp16` or `--bf16`) and half precision evaluation (`--fp16_full_eval` or `--bf16_full_eval`) can only be used on CUDA devices."
             )
 
+        if is_torch_available() and is_torch_tf32_available():
+            torch.backends.cuda.matmul.allow_tf32 = True if self.tf32 else False
```

Comment:
So here we should only change that boolean if the value set was not None. If the value is `True`, there should be an error if `is_torch_tf_32_available()` is `False` so the user is not surprised if they don't get what they want.

Reply:
thanks for this great feedback, Sylvain. Please have another look.